Rx/v2/src/rxcpp/operators/rx-zip.hpp

```diff
@@ -200,55 +241,120 @@ struct zip : public operator_base<rxu::value_type_t<zip_traits<Coordination, Sel
     }
```

Comment:
Trying to implement an operator in a new way..

```struct zip : public operator_base```
^^
Is that the preferred way to create an operator?